### PR TITLE
Activity Log store - part 4 - Add SQL methods and tests for RewindStatus

### PR DESCRIPTION
**Pull request by planarvoid**
- add storing of rewind status per site on fetch + add get method to retrieve rewind status
- move builder to the SqlUtils class

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/persistence/ActivityLogSqlUtils.kt [hunk 1]
```diff
@@ -117,4 +161,37 @@ class ActivityLogSqlUtils
                     actor)
         }
     }
+
+    @Table(name = "RewindStatus")
+    data class RewindStatusBuilder(@PrimaryKey
+                                   @Column private var mId: Int? = null,
```
[the remark this answers is not in the document]
Comment (planarvoid) [reply fluxc/src/main/java/org/wordpress/android/fluxc/persistence/ActivityLogSqlUtils.kt]: I don't think so, I'll revert the change

**Comment by aforcier** [comment]
This is from the previous PR, but @0nko had an issue that I think stems from this:

https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/762/files#diff-c5cd75ac6dae0131ec88212fc191c682R37

```
android.database.sqlite.SQLiteConstraintException: UNIQUE constraint failed: ActivityLog._id (code 1555)
        at android.database.sqlite.SQLiteConnection.nativeExecuteForChangedRowCount(Native Method)
        at android.database.sqlite.SQLiteConnection.executeForChangedRowCount(SQLiteConnection.java:735)
        at android.database.sqlite.SQLiteSession.executeForChangedRowCount(SQLiteSession.java:754)
        at android.database.sqlite.SQLiteStatement.executeUpdateDelete(SQLiteStatement.java:64)
        at android.database.sqlite.SQLiteDatabase.updateWithOnConflict(SQLiteDatabase.java:1653)
        at android.database.sqlite.SQLiteDatabase.update(SQLiteDatabase.java:1599)
```

The problem is that we're matching by `ACTIVITY_ID`, but then updating the entire model in the database if a match is found. It's possible for the `_id` field of the `ActivityLog` table to be different (say, a new fetch happened), and when we update the database we get a conflict with another entry that was already using that id.

The solution is to exclude the id from the updated fields with a custom `InsertMapper`. We have a pattern for this:

https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/4e7424082c191860a5ab8fc0f9f740b68e4f6145/fluxc/src/main/java/org/wordpress/android/fluxc/persistence/CommentSqlUtils.java#L46-L48

File: fluxc/src/main/java/org/wordpress/android/fluxc/persistence/ActivityLogSqlUtils.kt [hunk 2]
```diff
@@ -58,8 +58,9 @@ class ActivityLogSqlUtils
             WellSql.update(RewindStatusBuilder::class.java)
                     .where()
                     .equals(RewindStatusTable.ID, existingRewindStatus.id)
+                    .equals(ActivityLogTable.LOCAL_SITE_ID, existingRewindStatus.localSiteId)
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/persistence/ActivityLogSqlUtils.kt]: Should be `RewindStatusTable.LOCAL_SITE_ID`.

**Comment by aforcier** [comment]
This looks good to me - @0nko can you confirm you're not seeing that SQL error anymore?

**Comment by 0nko** [comment]
Yup, the problem's gone 👍.

**Comment by planarvoid** [comment]
@aforcier can you merge this if you're ok with it? :-)

**Comment by aforcier** [comment]
@planarvoid I was leaving it to @0nko to verify functionality since I mostly concentrated on FluxC patterns and gotchas and the like, and he's been working on the UI. Are you all set @0nko?

**Comment by 0nko** [comment]
I haven't implemented the rewind functionality in the UI yet since the 1st step is to have a complete read-only view functionality. But from what I've seen it seems 💯.

:shipit:

**Pull request merged**